src/player.js

```diff
@@ -1303,7 +1303,12 @@ export default class Player extends FakeEventTarget {
         this._engine.selectAudioTrack(track);
       } else if (track instanceof TextTrack) {
         this._resetTextDisplay();
-        if (track.language === OFF) {
+        const activeTracks = this.getActiveTracks();
```

Comment:
the issue regarding FEC-1111 can be fix without accessing the cache just by reverse the order of the condition 
`const currentOrConfiguredTextLang =
  this._getLanguage<TextTrack>(this._getTextTracks(), playbackConfig.textLanguage, activeTracks.text) ||
  this._playbackAttributesState.textLanguage;`